qutebrowser/browser/webengine/webenginesettings.py

```diff
@@ -244,6 +244,44 @@ def _init_profiles():
         private_profile.setSpellCheckEnabled(True)
 
 
+def inject_userscripts():
+    """Register user javascript files with the global profiles."""
+    # The greasemonkey metadata block support in qtwebengine only starts at 5.8
+    # Otherwise have to handle injecting the scripts into the page at very
+    # early load, probs same place in view as the enableJS check.
+    if not qtutils.version_check('5.8'):
+        return
+
+    # Since we are inserting scripts into profile.scripts they won't
+    # just get replaced by new gm scripts like if we were injecting them
+    # ourselves so we need to remove all gm scripts, while not removing
+    # any other stuff that might have been added. Like the one for
+    # stylsheets.
+    # Could either use a different world for gm scripts, check for gm metadata
+    # values (would mean no non-gm userscripts), or check the code for
+    # _qute_script_id
+    for profile in [default_profile, private_profile]:
+        scripts = profile.scripts()
+        for script in scripts.toList():
+            if script.name().startswith("GM-"):
+                log.greasemonkey.debug('removing script: {}'
+                                       .format(script.name()))
+                scripts.remove(script)
+
+    for profile in [default_profile, private_profile]:
+        scripts = profile.scripts()
+        greasemonkey = objreg.get('greasemonkey')
+        for script in greasemonkey.all_scripts():
+            new_script = QWebEngineScript()
```

Comment:
(Switching back to GitHub comments because there probably shouldn't be much left now, and I don't expect any changes from you anymore as I already began merging this in)

Is it intended that you never set the injection point here, i.e. all scripts are Deferred? Did we have timing issues when setting it correctly?

Reply:
This bit is for recent qwebengine which supports parsing the greasemonkey attributes by itself. The better question would be why am I setting RunsOnSubFrames explicitly? And that's because whatever QT version I have doesn't see to parse noframes. It does parse run-at fine though.

Reply:
Ah, right! I found this in Qt's sources which explains it:

```cpp
// FIXME: Scripts don't run in subframes by default. If we would like to
// support @noframes rule, we have to change the current default behavior.
// static const base::StringPiece kNoFramesDeclaration("// @noframes");
```